### PR TITLE
Refactor/addon

### PR DESCRIPTION
**Pull request by fsmanuel**
@nolanlawson don't merge yet. I'll add tests. Against what ember-data versions do you want the tests to run?

A short summary:
- [x] PouchDB shim
- [x] pouch adapter and serializer
- [x] pouch model (ships with the `rev` attribute)
- [x] README update

Any feedback is welcome.

### CODE REVIEW

**Comment by nolanlawson** [comment]
Hm, I guess the latest version of ember-data? I'm perfectly happy to drop support for older versions as well, if it makes your life easier. Nice work! This is looking fantastic. :tada: :balloon: :tada:

**Comment by fsmanuel** [comment]
@nolanlawson can you activate travis builds?

File: .npmignore [hunk 1]
```diff
@@ -0,0 +1,14 @@
+bower_components/
+tests/
+tmp/
+dist/
```
Comment (nolanlawson) [inline .npmignore]: adding `dist`/ to npmignore could cause problems for anybody who's `npm install`ing but then using the dist directory... would prefer to leave this in!

File: addon/adapters/pouch.js [hunk 2]
```diff
@@ -1,51 +1,68 @@
...
+  onChange: function (change) {
+    // If relational_pouch isn't initialized yet, there can't be any records
+    // in the store to update.
+    if (!this.db.rel) { return; }
+
+    var obj = this.db.rel.parseDocID(change.id);
+    // skip changes for non-relational_pouch docs. E.g., design docs.
+    if (!obj.type || !obj.id || obj.type === '') { return; }
+
+    var store = this.container.lookup('store:main');
```
Comment (nolanlawson) [inline addon/adapters/pouch.js]: Seems you removed the `store.getById` logic. Any reason for that?

File: addon/adapters/pouch.js [hunk 3]
```diff
@@ -184,8 +209,7 @@ exports["default"] = DS.RESTAdapter.extend({
   deleteRecord: function (store, type, record) {
     this._init(type);
     var data = this._recordToData(store, type, record);
-    return this.db.rel.del(type.typeKey, data).then(function () {
-      return ''; // ember doesn't like getting a json response of {deleted: true}
-    });
+    return this.db.rel.del(type.typeKey, data)
+      .then(extractDeleteRecord);
```
Comment (nolanlawson) [inline addon/adapters/pouch.js]: suppose the Ember bug noted in the comment has been fixed by now? Else I don't see any harm in keeping it in.

File: blueprints/ember-pouch/index.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+'use strict';
+
+module.exports = {
+  normalizeEntityName: function() {},
+
+  afterInstall: function() {
+    this.addBowerPackagesToProject([
+      { name: 'pouchdb', target: '~3.2.0' },
```
Comment (nolanlawson) [inline blueprints/ember-pouch/index.js]: this should really be `^3.5.0` by now.

File: blueprints/ember-pouch/index.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+      { name: 'relational-pouch', target: '~1.2.0'}
```
Comment (nolanlawson) [inline blueprints/ember-pouch/index.js]: ditto for this; using a `^` is totally safe because I follow semver. :)

**Comment by nolanlawson** [comment]
@fsmanuel Just turned on Travis, and verified a green run in https://github.com/nolanlawson/ember-pouch/pull/57.

This improves ember-pouch in so many different ways, I don't even know where to start. The new tests are amazing, and the new documentation is very much appreciated. I'm going to publish this today.

My only minor critique is that this doesn't seem to add any serious test cases - as far as I can tell, the PhantomJS tests are just testing JSHint. The direction I would like this plugin to go in is

1. to run the Ember Data test suite, and
2. to test syncing with a live CouchDB (this can be done in Travis; we already do it [in the PouchDB Travis tests](https://github.com/pouchdb/pouchdb/blob/3bcdefc88ae728b4fe1e8d60a051e403db262345/.travis.yml#L6-L7).

In general I admit that it's been my fault that I haven't been very rigorous in enforcing any tests for this plugin. Most of that I can chalk up to my lack of knowledge of Ember, although it's still no excuse. Luckily the Ember community has overwhelmed me with its ability to step up and make this plugin awesome, so I've been happy to just sit back and watch it happen. :)

In any case, you are now a contributor, although please do continue opening pull requests unless it's for minor fixes. I intend to do the same, even though this is my repo. :)

**Comment by nolanlawson** [comment]
Merged, although I reverted the README changes temporarily until we publish. Also, I'm going to wait a bit before publishing, until I've gotten some more feedback.

**Comment by nolanlawson** [comment]
Thanks again for the contribution; this is flippin' amazing!! :dancer:

**Comment by fivetanley** [comment]
ZOMG :tada:

**Comment by fsmanuel** [comment]
@nolanlawson hmm seems there is a bug with the testing. i added more than beta/release/canary. Maybe travis isn't picking them up because of the scenario name... when i run `ember try:testall` local i get
```bash
------ RESULTS ------

default: PASS
legacy ember-data 1.0.0.beta.12: FAIL
legacy ember-data 1.0.0.beta.14.1: FAIL
legacy ember-data 1.0.0.beta.15: PASS
legacy ember-data 1.0.0.beta.16.1: PASS
ember-release: PASS
ember-beta: PASS
ember-canary: PASS
```

We should open a new issue to address the failing scenarios and a testing story.

Beside testing we should look at @tchak PR where he did some amazing refactoring. That gives us the opportunity start adding tests and improve the code quality.

**Pull request merged**